libraries/Microsoft.Bot.Builder.Core.State/IStateStorageProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.State
+{
+    public interface IStateStorageProvider
```

Comment:
>Provider [](start = 34, length = 8)

Do we REALLY need the word "Provider" at the end of this interface and therefore every class the implements it? This is overly verbose for a JS developer.

Reply:
If you're asking from the .NET side, the "provider" pattern (and naming) is pretty well established there and I would suggest keeping it (though I'm not looking to die on any hill for it), but if you don't think the name helps on the JS then I don't think you need to use it. I think this is one of those things where differing for the ecosystem could make total sense.

Reply:
Just feels overly wordy for JS and StateStorage is pretty descriptive in itself

---
In reply to: [179843934](https://github.com/Microsoft/botbuilder-dotnet/pull/376#discussion_r179843934) [](ancestors = 179843934)

Reply:
I gotcha, again, I think it's totally fair to drop 'Provider' in JS. I only think we need to strive for architecture/concept alignment and naming/other styles should drift towards ecosystem patterns, but that's my opinion and if you guys, as the owners, don't agree I will of course rename back to just `IStateStorage`.

P.S. When it comes to wordy, .NET ain't go nothin' Java. 😆